UserGuide.md

```diff
@@ -378,10 +378,64 @@ This approach is practical only if the array appears as a top-level element or i
 
 ### Built-in Serializers and Deserializers
 
-Gson has built-in serializers and deserializers for commonly used classes whose default representation may be inappropriate, for instance
+Gson has built-in serializers and deserializers for commonly used classes whose default representation may be inappropriate.
+Gson provides built-in serializers and deserializers for basic Java types (e.g., primitive types, strings, arrays) as well as some commonly used classes like `Date`, `BigInteger`, `BigDecimal`, and more. When you use Gson to convert Java objects to JSON or JSON to Java objects, these built-in serializers and deserializers handle the conversion automatically for these basic types and classes.
 
-* `java.net.URL` to match it with strings like `"https://github.com/google/gson/"`
-* `java.net.URI` to match it with strings like `"/google/gson/"`
```

Comment:
Maybe it would be good to include `URL` and `URI` in the list below again with a description like "Serialized as their string representation" or similar, since these types might also be somewhat common.